echidna/limit/chi_squared.py

```diff
@@ -147,20 +124,23 @@ def pearson_chi_squared(observed, expected):
         events
 
     Raises:
-      ValueError: If either array contains a bin with content <= 0.0
+      ValueError: If arrays are different lengths.
 
     Returns:
       float: Calculated Pearson's chi squared
     """
-    correct_bin_content, message = check_bin_content(observed)
-    if not correct_bin_content:
-        raise ValueError("chi_squared.pearson_chi_squared: observed " +
-                         message)
-    correct_bin_content, message = check_bin_content(expected)
-    if not correct_bin_content:
-        raise ValueError("chi_squared.pearson_chi_squared: expected " +
-                         message)
-    return numpy.sum((observed-expected)**2 / expected)
+    if len(observed) != len(expected):
+        raise ValueError("Arrays are different lengths")
+    epsilon = 1e-34  # In the limit of zero
```

Comment:
No reference. It was an arbitrary choice. In the limit -> 0 the value of LL at that bin -> 0. It is only at exactly zero that LL is unphysical due to zero division. I chose a very small number just because some of our backgrounds have very low rates in bins in the ROI.